File: modules/sms-permission/ios/SmsPermissionModule.swift
import ExpoModulesCore

public class SmsPermissionModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SmsPermission")

    AsyncFunction("requestSmsPermission") { promise in
      promise.resolve(true)
    }

    Function("checkSmsPermission") {
      return false
    }

    Function("getSmsMessages") { (limit: Int?) in
      return []
    }

    Events("onPermissionResult")
  }
}

File: modules/sms-permission/src/SmsPermissionModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import { SmsPermissionModuleEvents, SmsMessage } from './SmsPermission.types';

declare class SmsPermissionModule extends NativeModule<SmsPermissionModuleEvents> {
  requestSmsPermission(): Promise<boolean>;
  checkSmsPermission(): boolean;
  getSmsMessages(limit?: number): Promise<SmsMessage[]>;
  startSmsListener(): Promise<boolean>;
  stopSmsListener(): Promise<boolean>;
  startForegroundService(): Promise<boolean>;
  stopForegroundService(): Promise<boolean>;
  showTransactionNotification(amount: string, merchant: string): Promise<boolean>;
}

export default requireNativeModule<SmsPermissionModule>('SmsPermission');

export type { SmsMessage, PermissionResultEvent, SmsReceivedEvent } from './SmsPermission.types';